go-controller/pkg/ovn/base_network_controller_pods.go

```diff
@@ -34,8 +34,11 @@ func (bnc *BaseNetworkController) allocatePodIPs(pod *kapi.Pod,
 	if !util.PodScheduled(pod) || util.PodWantsHostNetwork(pod) || util.PodCompleted(pod) {
 		return "", nil
 	}
-	// skip nodes that are not running ovnk (inferred from host subnets)
-	if bnc.lsManager.IsNonHostSubnetSwitch(switchName) {
+	// skip nodes that are not running ovnk (inferred from host subnets).
+	// flat L2 secondary networks may not have the subnet defined, which would
+	// result in them being identified by the `lsManager.IsNonHostSubnetSwitch`
+	// function.
+	if bnc.doesNetworkRequireIPAM() && bnc.lsManager.IsNonHostSubnetSwitch(switchName) {
```

Comment:
This is confusing. Why would we not skip `flat L2 secondary networks that have no subnet defined` in `nodes that are not running ovnk`? I think that needs to be addressed in the comment.